### PR TITLE
Debugger locals show wrong value for parameter changed and captured by closure in VB

### LINKED ISSUE #12219: Debugger locals show wrong value for parameter changed and captured by closure in VB
As detailed in #11949, we need to update https://github.com/dotnet/roslyn/blob/master/src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb#L1286 to reference the closure field instead of the original parameter location.

### PR DESCRIPTION
**Pull request by antonssonj**
Fix #12219

### CODE REVIEW

File: src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb [hunk 1]
```diff
@@ -1287,6 +1287,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
                     ' locals above.  We assert that we are seeing the latter.
                     Debug.Assert((variableKind = DisplayClassVariableKind.Parameter) OrElse
                         (variableKind = DisplayClassVariableKind.Me))
+
+                    If variableKind = DisplayClassVariableKind.Parameter AndAlso instance.Type.Name.StartsWith(StringConstants.DisplayClassPrefix, StringComparison.Ordinal) Then
```
Comment (antonssonj) [inline src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb]: GeneratedNames.GetKind does not include a LambdaDisplayClass so i have made that check in the if statement, maybe LambdaDisplayClass should be added to the GeneratedNameKind enum.

**Comment by tmat** [comment]
@KevinH-MS @cston

**Comment by crummel** [comment]
Build triggered. sha1 is merged.

**Comment by gafter** [comment]
@dotnet/roslyn-compiler Please review this community contribution.

File: src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb (hunk 1 above)
Comment (cston) [inline src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb]: Thanks for fixing this. Yes, please add `LambdaDisplayClass` to `GeneratedNameKind` and add a corresponding case to `GeneratedNames.GetKind()`.

**Comment by cston** [comment]
LGTM. Thanks @antonssonj.

**Comment by gafter** [comment]
LGTM. Thanks @antonssonj!

**Pull request merged**